packages/superset-ui-chart-companion/src/legend/WithLegend.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable sort-keys */
+import React, { CSSProperties, ReactNode, PureComponent } from 'react';
+import { ParentSize } from '@vx/responsive';
+// eslint-disable-next-line import/no-unresolved
+import * as CSS from 'csstype';
+
+const defaultProps = {
+  className: '',
+  height: 'auto' as number | string,
+  width: 'auto' as number | string,
+  position: 'top',
+};
+
+type Props = {
+  className: string;
+  width: number | string;
+  height: number | string;
+  legendJustifyContent?: 'center' | 'flex-start' | 'flex-end';
+  position: 'top' | 'left' | 'bottom' | 'right';
+  renderChart: (dim: { width: number; height: number }) => ReactNode;
+  renderLegend?: (params: { direction: string }) => ReactNode;
+} & Readonly<typeof defaultProps>;
+
+const LEGEND_STYLE_BASE: CSSProperties = {
+  display: 'flex',
+  flexGrow: 0,
+  flexShrink: 0,
+  fontSize: '0.9em',
+  order: -1,
+  paddingTop: '5px',
+};
+
+const CHART_STYLE_BASE: CSSProperties = {
+  flexBasis: 'auto',
+  flexGrow: 1,
+  flexShrink: 1,
+  position: 'relative',
+};
+
+class WithLegend extends PureComponent<Props, {}> {
+  static defaultProps = defaultProps;
+
+  getContainerDirection(): CSS.FlexDirectionProperty {
+    const { position } = this.props;
+
+    if (position === 'left') {
+      return 'row';
+    } else if (position === 'right') {
+      return 'row-reverse';
+    } else if (position === 'bottom') {
+      return 'column-reverse';
+    }
+
+    return 'column';
+  }
+
+  getLegendJustifyContent() {
+    const { legendJustifyContent, position } = this.props;
+    if (legendJustifyContent) {
+      return legendJustifyContent;
+    }
+
+    if (position === 'left' || position === 'right') {
+      return 'flex-start';
+    }
+
+    return 'flex-end';
+  }
+
+  render() {
+    const { className, width, height, position, renderChart, renderLegend } = this.props;
+
+    const isHorizontal = position === 'left' || position === 'right';
+
+    const style: CSSProperties = {
+      display: 'flex',
+      flexDirection: this.getContainerDirection(),
+    };
+
+    style.width = width;
```

Comment:
why not just set these when you define `style`? `const style = { ..., width, height };`

Reply:
ah true. delete the `if` condition that is no longer needed and forgot to look around.
will fix.